### ISSUE TITLE
convergency speed

### ISSUE THREAD

**Issue by yuntang** [twitter/torch-distlearn#4]
I follow the sample code and put distlearn into my RNN code. I find the training speed up to 3.5 times with 4 GPUs. However, the convergency speed is even slower than 1 GPU case. That means the script can process 3.5 times more training data with 4 GPUs, however it takes similar or longer time to achieve the same accuracy using 1 GPU. I have tuned the parameters: alpha (0.02 to 0.5) and tau (2~10 mini-batch updates) with different values. I also tried to synchronize the parameters (synchronizeParameters) in different ways: from every epoch (tens of hours) to a few of minutes. 
After those changes, the convergency speed is still slower than 1 GPU case. Any hint for the problem?

Thanks

Yun

**Comment by clementfarabet** [comment]
@yuntang , can you try to sync as often as possible? Like closer to every few batches/updates. A few minutes sounds like a lot. In general it takes a bit of tuning to get elastic averaging to work right.

**Comment by zakattacktwitter** [comment]
Hmm, I think synch is fine once an epoch, it just propagates node 0's
Params to all the others, any more often and you might as well turn the
other nodes off. Try more aggressive alphas, I've had ok results with
0.6-0.7 range. As with most things searching the hyper parameter space is
critical to getting good performance.

**Comment by yuntang** [comment]
@clementfarabet and @zakattacktwitter, thanks for the feedback. 
Do you think elastic averaging might work better on model initialized with single GPU?

**Comment by zakattacktwitter** [comment]
Not sure what you mean by that, the very first call to synch Params before
training begins should put identical parameter values on each node. Once
trying starts they diverge. Are you starting with a call to synch?

**Comment by yuntang** [comment]
Sorry for the confusion.
I mean it might be useful to start the parallel training from a seed model which is trained with one GPU using small amount of training data. 
I synchronize parameters for different nodes at the beginning of the training and also in the middle of training. 
My latest observation is if the parameters are synchronized every ten mins instead of one epoch (tens of hours) during training, the parallel training (with 2 GPUs) is converged slightly faster than the one GPU case.  I think the training speed could be improved further if the right training recipe could be found.  Thanks for the suggestion.
